Userland/Utilities/realsize.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2024, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/CharacterTypes.h>
+#include <AK/Checked.h>
+#include <AK/NumberFormat.h>
+#include <AK/Optional.h>
+#include <AK/StringView.h>
+#include <AK/Types.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
+
+static Optional<u64> use_integer_with_char_suffix(StringView argument, size_t suffix_length)
+{
+    if (suffix_length >= argument.length())
+        return {};
+    argument = argument.substring_view(0, argument.length() - suffix_length);
+
+    Optional<u64> numeric_optional = argument.to_number<u64>();
+    return numeric_optional;
+}
+
+static ErrorOr<size_t> handle_number_with_one_char_suffix(AK::HumanReadableBasedOn human_readable_based_on, StringView argument)
+{
+    VERIFY(human_readable_based_on == AK::HumanReadableBasedOn::Base2
+        || human_readable_based_on == AK::HumanReadableBasedOn::Base10);
+    auto numeric_value_without_suffix = use_integer_with_char_suffix(argument, 1).value();
+    u64 suffix_multiplier = 1;
+    auto suffix = argument[argument.length() - 1];
+
+    switch (suffix) {
```

Comment:
Would it be worth making the suffixes case insensitive here. Given that we require the user to specify whether they want `--base2` or `--base10` for a 1 char suffix, it feels like `realsize --base10 1m` should be able to return a result.

Reply:
The problem with this, is that people could then specify "kiB" or "mB", which is weird in my opinion.
What do you think about this?

Reply:
How about only treating single character suffixes as case-insensitive?

Reply:
How strongly do you feel about case-insensitivity here? SI prefixes are standardized, and `M` and `m` do mean separate things etc. <https://en.wikipedia.org/wiki/Kilobyte>: "The internationally recommended unit symbol for the kilobyte is kB."

Reply:
I can't say I feel particularly strongly about it to be honest.

Reply:
I don't feel strongly about this either. I just don't want to add more complexity here. So the neatest and simplest way should be taken IMHO.